scripts/test.py

```diff
@@ -95,31 +98,27 @@ def hack_emsdk(marker, replacement):
 
 
 def test_lib_building(emcc, use_asmjs_optimizer):
-  def test_build(args, expected=None, unexpected=None):
+  cache_building_messages = ['generating system library: ']
+
+  def test_build(args, expected):
+    if expected:
+      expected = cache_building_messages
+      unexpected = []
+    else:
+      expected = []
+      unexpected = cache_building_messages
     checked_call_with_output(emcc + ' hello_world.c' + args,
                              expected=expected,
                              unexpected=unexpected,
                              stderr=subprocess.STDOUT)
 
-  # by default we ship libc, struct_info, and the asm.js optimizer, as they
-  # are important for various reasons (libc takes a long time to build;
-  # struct_info is a bootstrap product so if the user's setup is broken it's
-  # confusing; the asm.js optimizer is a native application so it needs a
-  # working native local build environment). otherwise we don't ship every
-  # single lib, so some building is expected on first run.
-
-  unexpected_system_libs = ['generating system library: libc.',
-                            'generating system asset: optimizer']
-  if use_asmjs_optimizer:
-    unexpected_system_libs += ['generating system asset: generated_struct_info.json']
-
-  first_time_system_libs = ['generating system library: libdlmalloc.']
-
-  test_build('', expected=first_time_system_libs,
-             unexpected=unexpected_system_libs)
-  test_build(' -O2', unexpected=unexpected_system_libs + first_time_system_libs)
-  test_build(' -s WASM=0', unexpected=unexpected_system_libs + first_time_system_libs)
-  test_build(' -O2 -s WASM=0', unexpected=unexpected_system_libs + first_time_system_libs)
+  # The emsdk ships all system libraries so we don't expect to see any
+  # cache population unless we explicly --clear-cache.
+  test_build('', expected=False)
+  check_call(emcc + ' --clear-cache')
```

Comment:
this is maybe excessive - CI here will wait on building all of libc. maybe just delete say `libdlmalloc`?

Reply:
The problem is that each version of SDK now has it own cache... so I'd need to figure out where each libc is.  For the upstream SDK I know this because of `EM_CACHE`, but for fastcomp I'd need to derive the cache directory somehow.

Reply:
Ok, how about removing fastcomp cache testing then?

Reply:
Sure.. although building libc only takes less than a minute, no?    Does it it matter on the emsdk CI?

Reply:
Not a big deal, yeah. But it's nice if emsdk CI is fast as we want to avoid anything landing on emscripten while we tag. lgtm either way though.